mmv1/third_party/terraform/services/backupdr/data_source_backup_dr_backup_plan_test.go

```diff
@@ -78,3 +99,57 @@ data "google_backup_dr_backup_plan" "fetch-bp" {
   }
 `, context)
 }
+
+func testAccDataSourceGoogleBackupDRBackupPlan_csql(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_backup_dr_backup_vault" "my-backup-vault-csql" {
+    location ="us-central1"
+    backup_vault_id    = "bv-%{random_suffix}"
+    description = "This is a backup vault built by Terraform for cloudsql."
+    backup_minimum_enforced_retention_duration = "100000s"
+    labels = {
+      foo = "bar1"
+      bar = "baz1"
+    }
+    annotations = {
+      annotations1 = "bar1"
+      annotations2 = "baz1"
+    }
+    force_update = "true"
+    force_delete = "true"
+    allow_missing = "true" 
+}
+
+
+resource "google_backup_dr_backup_plan" "csql-test" { 
+  location = "us-central1" 
+  backup_plan_id = "bp-test-%{random_suffix}"
+  resource_type= "sqladmin.googleapis.com/Instance"
+  backup_vault = google_backup_dr_backup_vault.my-backup-vault-csql.name
+  depends_on=[ google_backup_dr_backup_vault.my-backup-vault-csql ]
+  lifecycle {
+    ignore_changes = [backup_vault]
+  }
+  log_retention_days = 4
+  backup_rules {
+    rule_id = "rule-1"
+    backup_retention_days = 5
+    standard_schedule {
+      recurrence_type = "HOURLY"
+      hourly_frequency = 6
+      time_zone = "UTC"
+      backup_window{
+        start_hour_of_day = 0
+        end_hour_of_day = 24
+      }
+    }
+  }
+}
+
+data "google_backup_dr_backup_plan" "fetch-bp" {
+  location =  "us-central1"
+  backup_plan_id="bp-test-%{random_suffix}"
```

Comment:
```suggestion
  backup_plan_id= google_backup_dr_backup_plan.csql-test.backup_plan_id
```

Let's use reference instead of hardcoding the id